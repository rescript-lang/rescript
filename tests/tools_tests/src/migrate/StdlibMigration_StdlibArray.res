let ba1 = [1, 2, 3, 4, 5]->Array.copyAllWithin(~target=2)
let ba2 = Array.copyAllWithin([1, 2, 3, 4, 5], ~target=2)

let b1 = [1, 2, 3, 4]->Array.copyWithinToEnd(~target=0, ~start=2)
let b2 = Array.copyWithinToEnd([1, 2, 3, 4], ~target=0, ~start=2)

let c1 = [1, 2, 3]->Array.fillAll(0)
let c2 = [1, 2, 3, 4]->Array.fillToEnd(9, ~start=1)

let d1 = [1, 2, 1, 2]->Array.indexOfFrom(2, 2)
let d2 = Array.indexOfFrom([1, 2, 1, 2], 2, 2)

let e1 = ["a", "b"]->Array.joinWith("-")
let e2 = [1, 2]->Array.joinWithUnsafe(",")

let f1 = [1, 2, 3, 4]->Array.sliceToEnd(~start=2)

let g1 = [1, 2]->Array.lastIndexOfFrom(1, 1)

let h1 = [1, 2]->Array.unsafe_get(1)
