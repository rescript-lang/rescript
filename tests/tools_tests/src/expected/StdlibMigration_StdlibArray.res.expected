let ba1 = [1, 2, 3, 4, 5]->Array.copyWithin(~target=2, ~start=0)
let ba2 = Array.copyWithin([1, 2, 3, 4, 5], ~target=2, ~start=0)

let b1 = [1, 2, 3, 4]->Array.copyWithin(~target=0, ~start=2)
let b2 = Array.copyWithin([1, 2, 3, 4], ~target=0, ~start=2)

let c1 = [1, 2, 3]->Array.fill(0)
let c2 = [1, 2, 3, 4]->Array.fill(9, ~start=1)

let d1 = [1, 2, 1, 2]->Array.indexOf(2, ~from=2)
let d2 = Array.indexOf([1, 2, 1, 2], 2, ~from=2)

let e1 = ["a", "b"]->Array.join("-")
let e2 = [1, 2]->Array.joinUnsafe(",")

let f1 = [1, 2, 3, 4]->Array.slice(~start=2)

let g1 = [1, 2]->Array.lastIndexOf(1, ~from=1)

let h1 = [1, 2]->Array.getUnsafe(1)

